alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -92,14 +84,12 @@ essentials::IdentifierConstPtr BasicBehaviour::getOwnId() const
  */
 bool BasicBehaviour::stop()
 {
-    {
-        std::lock_guard<std::mutex> lck(_runLoopMutex);
-        setSignalState(SignalState::STOP);
-        setStopCalled(true);
-    }
-    if (_behaviour->isEventDriven()) {
-        _runCV.notify_all();
+    if (!isStopCalled()) {
+        _engine->editScheduler().schedule(std::bind(&BasicBehaviour::doTermination, this));
     }
+
+    setSignalState(SignalState::STOP);
+    setStopCalled(true);
```

Comment:
Too many trackers for state. As @veerajsk  you said before, we definitely need to clean this up in a different pr.